### ISSUE TITLE
WPF DataGrid accessibility issues (interactive element not focusable by default, sort order not announced by narrator, editable cells not announced by narrator)

### ISSUE THREAD

**Issue by sfreed141** [dotnet/wpf#5620]
* .NET Core Version:
output from `dotnet --info`:
```
.NET SDK (reflecting any global.json):
 Version:   6.0.100
 Commit:    9e8b04bbff

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22000
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\6.0.100\

Host (useful for support):
  Version: 6.0.0
  Commit:  4822e3c3aa
```
* Windows version: (`winver`): Version 21H2 (OS Build 22000.258)
* Does the bug reproduce also in WPF for .NET Framework 4.8?: Yes
* Is this bug related specifically to tooling in Visual Studio (e.g. XAML Designer, Code editing, etc...)? no

 
 
 **Problem description:**
 This is a follow up for internal work item https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1423483.
In VS we sometimes receive accessibility issues that repro with the default WPF behavior for DataGrid. It would be great if these problems would be addressed in WPF itself so the default behavior is accessible. To test, just launch the attached repro project (it's just an empty DataGrid with a few rows defined in the code-behind) and use tab to navigate to the relevant control. Shortcut to enable narrator is ctrl+win+enter.

 **Actual behavior:** 
1. By default the DataGridColumnHeader is not focusable (i.e. cannot navigate to it using keyboard controls), even when the column can be sorted.
2. Narrator does not announce sort order of the column (when DataGridColumnHeader is focused).
3. Narrator does not announce editable DataGrid cells as editable (this may be a bug on narrator). Can confirm the field is editable by pressing F2.

 **Expected behavior:**
  Accessibility reasons corresponding to the above problems
1. Interactive elements must be focusable using keyboard controls
2. Narrator should announce the sort order
3. Narrator should announce editable fields

 **Minimal repro:**
[WpfDataGrid.zip](https://github.com/dotnet/wpf/files/7462767/WpfDataGrid.zip)

**Comment by anjalisheel-wpf** [comment]
By default the DataGridColumnHeader is not focusable (i.e. cannot navigate to it using keyboard controls), even when the column can be sorted. - this is working as expected, check attached video for the reference.

Narrator does not announce editable DataGrid cells as editable - this is as per the design.

Narrator does not announce sort order of the column (when DataGridColumnHeader is focused). - this can be addressed.

https://user-images.githubusercontent.com/85492722/147400337-5781b587-7946-4ee5-8626-d29a272a79e3.mp4

**Comment by anjalisheel-wpf** [comment]
@sfreed141 

This is a narrator bug because the narrator announces automation name property. To announce sorting order, it must be looking for some property which the narrator team has to address as that is the additional information apart from automation name property.

**Comment by anjalisheel-wpf** [comment]
@sfreed141

Narrator does not announce sort order of the column (when DataGridColumnHeader is focused). - This is a narrator bug because the narrator announces automation name property. To announce sorting order, it must be looking for some property which the narrator team has to address as that is the additional information apart from automation name property.
